src/coreclr/jit/llvm.cpp

```diff
@@ -1081,7 +1072,32 @@ llvm::Value* Llvm::buildUserFuncCall(GenTreeCall* call)
     for (GenTreeCall::Use& use : call->Args())
     {
         lastArg = use.GetNode()->AsPutArgType();
-        argVec.push_back(consumeValue(lastArg->gtGetOp1(), getLlvmTypeForCorInfoType(lastArg->GetCorInfoType(), lastArg->GetClsHnd())));
+
+        if (lastArg->gtGetOp1()->OperIs(GT_FIELD_LIST))
+        {
+            assert(lastArg->GetCorInfoType() == CorInfoType::CORINFO_TYPE_VALUECLASS);
+
+            // alloca a type using the lastArg type
+            Type* argLlvmType = getLlvmTypeForCorInfoType(lastArg->GetCorInfoType(), lastArg->GetClsHnd());
+
+            GenTree* argNode = lastArg->gtGetOp1();
+            Value*   argValue;
+            if (lastArg->gtGetOp1()->OperIs(GT_FIELD_LIST))
+            {
+                argValue = buildFieldList(argNode->AsFieldList(), argLlvmType);
+            }
+            else
+            {
+                argValue = consumeValue(argNode, argLlvmType);
+            }
+
+            argVec.push_back(argValue);
+        }
+        else
+        {
+            argVec.push_back(consumeValue(lastArg->gtGetOp1(),
+                                          getLlvmTypeForCorInfoType(lastArg->GetCorInfoType(), lastArg->GetClsHnd())));
+        }
```

Comment:
```suggestion
        GenTree* argNode     = lastArg->gtGetOp1();
        Type*    argLlvmType = getLlvmTypeForCorInfoType(lastArg->GetCorInfoType(), lastArg->GetClsHnd());
        Value*   argValue;
        if (argNode->OperIs(GT_FIELD_LIST))
        {
            assert(lastArg->GetCorInfoType() == CORINFO_TYPE_VALUECLASS);
            argValue = buildFieldList(argNode->AsFieldList(), argLlvmType);
        }
        else
        {
            argValue = consumeValue(argNode, argLlvmType);
        }

        argVec.push_back(argValue);
```
Simplifying the code a bit.

Reply:
Missed that double use of `OperIs(GT_FIELD_LIST)`, thanks.  In https://github.com/dotnet/runtimelab/pull/2007/commits/df05ac468ac2df80292e0cd739f1d2dfe09dd27f